service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/DateFormatUtil.java

```diff
@@ -54,11 +68,39 @@ public static String toTextFormat(BaseDateTimeType baseDateTimeType) {
                     + String.format(FORMAT_TWO_DIGITS, baseDateTimeType.getMonth() + MONTH_PADDING);
             case DAY:
                 return baseDateTimeType.getYear() + "-"
-                    + String.format(FORMAT_TWO_DIGITS,  baseDateTimeType.getMonth() + MONTH_PADDING)
+                    + String.format(FORMAT_TWO_DIGITS, baseDateTimeType.getMonth() + MONTH_PADDING)
                     + "-" + String.format(FORMAT_TWO_DIGITS, baseDateTimeType.getDay());
             default:
                 return baseDateTimeType.toCalendar().toInstant().atZone(UK_ZONE_ID)
                     .format(HL7_SECONDS_HUMAN_READABLE);
         }
     }
+
+    public static DateTimeType toDateTypeTime(String effectiveTimeHl7Format) {
+        SimpleDateFormat formatter;
+        switch (effectiveTimeHl7Format.length()) {
+            case YEAR_PRECISION:
+                formatter = new SimpleDateFormat(HL7_DATE_YEAR_FORMAT, Locale.ENGLISH);
+                break;
+            case MONTH_PRECISION:
+                formatter = new SimpleDateFormat(HL7_DATE_MONTH_FORMAT, Locale.ENGLISH);
+                break;
+            case DAY_PRECISION:
+                formatter = new SimpleDateFormat(HL7_DATE_FORMAT, Locale.ENGLISH);
+                break;
+            default:
+                formatter = new SimpleDateFormat(HL7_DATETIME_FORMAT, Locale.ENGLISH);
+        }
```

Comment:
Wonderful example of when switch expressions would be useful!
```suggestion
        SimpleDateFormat formatter = switch (effectiveTimeHl7Format.length()) {
            case YEAR_PRECISION -> new SimpleDateFormat(HL7_DATE_YEAR_FORMAT, Locale.ENGLISH);
            case MONTH_PRECISION -> new SimpleDateFormat(HL7_DATE_MONTH_FORMAT, Locale.ENGLISH);
            case DAY_PRECISION -> new SimpleDateFormat(HL7_DATE_FORMAT, Locale.ENGLISH);
            default -> new SimpleDateFormat(HL7_DATETIME_FORMAT, Locale.ENGLISH);
        }
```

Alas, we're not on that version of java yet 😩

Reply:
look super nice, but, yes, we do not use that version of java.